src/Coordination/KeeperStorage.cpp

```diff
@@ -621,11 +610,11 @@ struct KeeperStorageSetRequestProcessor final : public KeeperStorageRequestProce
 
             auto itr = container.updateValue(request.path, [zxid, request] (KeeperStorage::Node & value)
             {
-                value.data = request.data;
                 value.stat.version++;
                 value.stat.mzxid = zxid;
                 value.stat.mtime = std::chrono::system_clock::now().time_since_epoch() / std::chrono::milliseconds(1);
                 value.stat.dataLength = request.data.length();
+                value.size_bytes += request.data.size() - value.data.size();
```

Comment:
What if this diff is negative? Probably we will get an overflow subtracting two unsigned integers https://www.cplusplus.com/reference/string/string/size/?

Reply:
fix.

Reply:
> 

That's really considerate.